ui/core-react/src/core-react/l10n/useTranslation.tsx

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as defaults from "../UiCore.json";
+import * as React from "react";
+import { UiCore } from "../UiCore";
+import { usePackageTranslation } from "./usePackageTranslation";
+
+/** Returns a translation function to localize package components.
+ * @internal
+ */
+export function useTranslation() {
+  const fallback = React.useCallback((key: string) => {
+    if (!UiCore.initialized) {
+      return undefined;
+    }
+
+    // eslint-disable-next-line deprecation/deprecation
+    return UiCore.translate(key);
```

Comment:
1. Why did you deprecate an internal function?
2. If this method is deprecated, what is the the future plan here? Do we want to remove this fallback?
3. Why did you deprecate the translate method only in core-react and not the other packages?

Reply:
- I deprecated it so we wouldn't use it by accident anywhere else other than in `useTranslation` for a fallback.
- The future plan is to see if we can make corresponding changes to all our other packages. Then we can deprecate  those static initializers like `UiCore.initialize()`.
- I planned to look into other packages separately, since this PR is already getting huge